cpp/velox/benchmarks/ParquetWriteBenchmark.cc

```diff
@@ -0,0 +1,352 @@
+    }
+
+    state.counters["rowgroups"] =
+        benchmark::Counter(rowGroupIndices_.size(), benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+    state.counters["columns"] =
+        benchmark::Counter(columnIndices_.size(), benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+    state.counters["batches"] =
+        benchmark::Counter(numBatches, benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+    state.counters["num_rows"] =
+        benchmark::Counter(numRows, benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+    state.counters["batch_buffer_size"] =
+        benchmark::Counter(kBatchBufferSize, benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1024);
+
+    state.counters["parquet_parse"] =
+        benchmark::Counter(elapseRead, benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+    state.counters["init_time"] =
+        benchmark::Counter(initTime, benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+    state.counters["write_time"] =
+        benchmark::Counter(writeTime, benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+  }
+};
+
+} // namespace gluten
+
+// GoogleBenchmarkVeloxParquetWriteCacheScanBenchmark usage
+// ./parquet_write_benchmark --threads=1 --file /mnt/DP_disk1/int.parquet --output file:/tmp/parquet-write
+// GoogleBenchmarkArrowParquetWriteCacheScanBenchmark usage
+// ./parquet_write_benchmark --threads=1 --file /mnt/DP_disk1/int.parquet --output /tmp/parquet-write
+int main(int argc, char** argv) {
+  initVeloxBackend();
+  uint32_t iterations = 1;
+  uint32_t threads = 1;
+  std::string datafile;
+  uint32_t cpu = 0xffffffff;
+  std::string output;
+
+  for (int i = 0; i < argc; i++) {
+    if (strcmp(argv[i], "--iterations") == 0) {
+      iterations = atol(argv[i + 1]);
+    } else if (strcmp(argv[i], "--threads") == 0) {
+      threads = atol(argv[i + 1]);
+    } else if (strcmp(argv[i], "--file") == 0) {
+      datafile = argv[i + 1];
+    } else if (strcmp(argv[i], "--cpu") == 0) {
+      cpu = atol(argv[i + 1]);
+    } else if (strcmp(argv[i], "--output") == 0) {
+      output = (argv[i + 1]);
+    }
+  }
+  std::cout << "iterations = " << iterations << std::endl;
+  std::cout << "threads = " << threads << std::endl;
+  std::cout << "datafile = " << datafile << std::endl;
+  std::cout << "cpu = " << cpu << std::endl;
+  std::cout << "output = " << output << std::endl;
+
+  gluten::GoogleBenchmarkVeloxParquetWriteCacheScanBenchmark bck(datafile, output);
+
+  benchmark::RegisterBenchmark("GoogleBenchmarkParquetWrite::CacheScan", bck)
+      ->Args({
+          cpu,
+      })
+      ->Iterations(iterations)
+      ->Threads(threads)
+      ->ReportAggregatesOnly(false)
+      ->MeasureProcessCPUTime()
+      ->Unit(benchmark::kSecond);
+
+  benchmark::Initialize(&argc, argv);
+  benchmark::RunSpecifiedBenchmarks();
+  benchmark::Shutdown();
+}
```

Comment:
Please add empty line at the end of file

Reply:
Done.